packages/x-data-grid/src/hooks/features/rows/useGridRowAriaAttributes.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { GridRowId } from '../../../models/gridRows';
+import { GridRowInternalHook } from '../../../models/configuration/gridRowConfiguration';
+import { selectedIdsLookupSelector } from '../rowSelection';
+import { useGridSelector } from '../../utils/useGridSelector';
+import { gridColumnGroupsHeaderMaxDepthSelector } from '../columnGrouping/gridColumnGroupsSelector';
+import { useGridPrivateApiContext } from '../../utils/useGridPrivateApiContext';
+
+export const useGridRowAriaAttributes: GridRowInternalHook = () => {
+  const apiRef = useGridPrivateApiContext();
+  const selectedIdsLookup = useGridSelector(apiRef, selectedIdsLookupSelector);
+  const headerGroupingMaxDepth = useGridSelector(apiRef, gridColumnGroupsHeaderMaxDepthSelector);
+
+  return React.useCallback(
+    (rowId: GridRowId, index: number) => {
+      const rowNode = apiRef.current.getRowNode(rowId);
```

Comment:
Fyi, some API methods are [basically just hidden selectors](https://github.com/mui/mui-x/blob/83c7554732b56d69abc1bb55a67046371084e2ae/packages/x-data-grid/src/hooks/features/rows/useGridRows.ts#L298). I'm not sure if you meant for this to be reactively bound though.

Reply:
`rowNode` is already fetched in `GridRow`, so you could pass it as a parameter here instead. It's not reactively bound in `GridRow` but it doesn't seem to have caused any bug as far as I know.

Reply:
I tend to choose primitives over objects as params since it increases flexibility and re-usability of the hooks/utils
in this particular case, I think it is fine to pass the row though, since the hook is only been used in one place